packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -0,0 +1,428 @@
+
+pragma solidity ^0.8.19;
+
+import { PausableUpgradeable } from '@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol';
+import { ReentrancyGuardUpgradeable } from '@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol';
+import { OwnableUpgradeable } from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import { IERC20 } from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import { INounsAuctionHouseV2 } from './interfaces/INounsAuctionHouseV2.sol';
+import { INounsToken } from './interfaces/INounsToken.sol';
+import { IWETH } from './interfaces/IWETH.sol';
+
+contract NounsAuctionHouseV2 is
+    INounsAuctionHouseV2,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable
+{
+    /// @notice A hard-coded cap on time buffer to prevent accidental auction disabling if set with a very high value.
+    uint56 public constant MAX_TIME_BUFFER = 1 days;
+
+    /// @notice The Nouns ERC721 token contract
+    INounsToken public immutable nouns;
+
+    /// @notice The address of the WETH contract
+    address public immutable weth;
+
+    /// @notice The duration of a single auction
+    uint256 public immutable duration;
+
+    /// @notice The minimum price accepted in an auction
+    uint192 public reservePrice;
+
+    /// @notice The minimum amount of time left in an auction after a new bid is created
+    uint56 public timeBuffer;
+
+    /// @notice The minimum percentage difference between the last bid amount and the current bid
+    uint8 public minBidIncrementPercentage;
+
+    /// @notice The active auction
+    INounsAuctionHouseV2.AuctionV2 public auction;
+
+    /// @notice Whether this contract is paused or not
+    /// @dev Replaces the state variable from PausableUpgradeable, to bit pack this bool with `auction` and save gas
+    bool public __paused;
+
+    /// @notice The Nouns price feed state
+    mapping(uint256 => SettlementState) settlementHistory;
+
+    constructor(
+        INounsToken _nouns,
+        address _weth,
+        uint256 _duration
+    ) {
+        nouns = _nouns;
+        weth = _weth;
+        duration = _duration;
+    }
+
+    /**
+     * @notice Initialize the auction house and base contracts,
+     * populate configuration values, and pause the contract.
+     * @dev This function can only be called once.
+     */
+    function initialize(
+        uint192 _reservePrice,
+        uint56 _timeBuffer,
+        uint8 _minBidIncrementPercentage
+    ) external initializer {
+        __Pausable_init();
+        __ReentrancyGuard_init();
+        __Ownable_init();
+
+        _pause();
+
+        reservePrice = _reservePrice;
+        timeBuffer = _timeBuffer;
+        minBidIncrementPercentage = _minBidIncrementPercentage;
+    }
+
+    /**
+     * @notice Settle the current auction, mint a new Noun, and put it up for auction.
+     */
+    function settleCurrentAndCreateNewAuction() external override whenNotPaused {
```

Comment:
@eladmallel `whenNotPaused` and `whenPaused` are not using the new `__paused` variable

Reply:
ignore!